### PR TITLE
Add configurations for challenge answer validation

### PR DESCRIPTION
**Pull request by GANGANI**
**PR contains:**
- Configurations of Identity.xml.j2 for challenge question validation
- Identity event constants related to challenge question validation
- Extended classes of IdentityEventClientException class

**Documentation:** [Challenge answer validation](https://docs.google.com/document/d/1Mk0iO4b_22xuQO5A-I-iFSfgdSHnuhk-ii-z41J0bao/edit?usp=sharing)

**Related PR:** https://github.com/wso2-extensions/identity-governance/pull/382

### When should this PR to be merged
This PR should be merged prior to the https://github.com/wso2-extensions/identity-governance/pull/382

### CODE REVIEW

File: components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/IdentityEventConstants.java [hunk 1]
```diff
@@ -202,7 +204,8 @@ private EventProperty(){}
         public static final String SESSION_CONTEXT = "sessionContext";
         public static final String PARAMS = "params";
         public static final String AUTHENTICATION_STATUS = "authenticationStatus";
-
+        public static final String USER_CHALLENGE_ANSWERS = "userChallengeAnswers";
+        public static final String USER_OLD_CHALLENGE_ANSWERS = "userNewChallengeAnswers";
```
Comment (somindatommy) [inline components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/IdentityEventConstants.java:208]: Shall we change the identifier to match the value?

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 2]
```diff
@@ -926,6 +926,10 @@
             <RecoveryLink>{{identity_mgt.password_reset_by_admin.enable_emailed_link_based_reset}}</RecoveryLink>
         </AdminPasswordReset>
         <CallbackRegex>{{identity_mgt.recovery.callback_url}}</CallbackRegex>
+        <ChallengeQuestion>
```
Comment (somindatommy) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:929]: `<Question>` tag is also for the challenge questions right? Why do we need a new tag?

**Comment by somindatommy** [comment]
@GANGANI If there are some other PRs related to this PR, Shall we add them in the description? WDYT?

File: components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/IdentityEventConstants.java (hunk 1 above)
Comment (GANGANI) [reply components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/IdentityEventConstants.java]: Fixed https://github.com/wso2/carbon-identity-framework/pull/2919/commits/ea7af90737d21f62569a67c50cfdc58d7f96e025

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 (hunk 2 above)
Comment (GANGANI) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2]: Fixed https://github.com/wso2/carbon-identity-framework/pull/2919/commits/ea7af90737d21f62569a67c50cfdc58d7f96e025

**Comment by GANGANI** [comment]
> @GANGANI If there are some other PRs related to this PR, Shall we add them in the description? WDYT?

I'll mentioned the related PRs, once I send those PRs after updating the released framework version.

**Pull request merged**